### PR TITLE
Entry Emulator - Summary Block

### PR DESCRIPTION
**Pull request by ikesau**
Some quick and dirty groundwork for the entry emulator (hard-coding entry slugs) plus migration code for the summary block.

I made the `SDGTableOfContents` component slightly more generic to suit these two usecases.

This doesn't work for summary headings with data values in them, or [summaries that have images in them](https://ourworldindata.org/soil-lifespans). In the interests of velocity I nominate we track those (via the error messages I've added) and manually update them.

[Datasette view of all our entries with summary blocks](https://owid-datasette-y43qr.ondigitalocean.app/owid?sql=select+%22https%3A%2F%2Fourworldindata.org%2F%22%7C%7Cslug+as+URL+from+posts+where+content+like+%22%25owid%2Fsummary%25%22+and+status+%3D+%22publish%22)

image
image

### CODE REVIEW

**Comment by mlbrgl** [comment]
- [ ] is the branch up-to-date with master?
- [ ] what problem is being solved here?
- [ ] what solution has been chosen?
- [ ] are there other solutions?
- [ ] is the scope clear, well-defined and small?
- [ ] could a flow chart / sequence diagram / architecture diagram help?
- [ ] what is the happy path?
- [ ] could the code be easier to read and understand?
- [ ] can I help increase our confidence in the implementation by contributing tests?
- [ ] should some errors be made more prominent?
- [ ] what is missing, what should have been added / done / changed but hasn't?

File: db/model/Gdoc/htmlToEnriched.ts [hunk 1]
```diff
@@ -596,6 +614,92 @@ function finishWpComponent(
...
+                }
+                return { errors: [], content: [callout] }
+            }
+
+            const contentIsList =
+                content.content.length === 1 &&
+                "type" in content.content[0] &&
+                content.content[0].type === "list"
+            if (contentIsList) {
+                const listItems = get(content, ["content", 0])
```
Comment (mlbrgl) [inline db/model/Gdoc/htmlToEnriched.ts:641]: it seems like "items" is missing and so list-based summary blocks (e.g. https://ourworldindata.org/working-hours) return an empty block (yet visible and expandable).

```suggestion
                const listItems = get(content, ["content", 0, "items"])
```

File: db/model/Gdoc/htmlToEnriched.ts [hunk 2]
```diff
@@ -596,6 +614,92 @@ function finishWpComponent(
                 }
             } else return { ...content, errors }
         })
+        .with("owid/summary", () => {
+            // Summaries can either be lists of anchor links, or paragraphs of text
+            // If it's a paragraph of text, we want to turn it into a callout block
+            // If it's a list of anchor links, we want to turn it into a toc block
```
Comment (mlbrgl) [inline db/model/Gdoc/htmlToEnriched.ts:620]: When a summary block is both text and list (e.g. https://ourworldindata.org/vaccination), both conditions get skipped below and the summary gets stripped out of the output.

**Review by mlbrgl — CHANGES_REQUESTED** [review]

File: db/model/Gdoc/htmlToEnriched.ts (hunk 1 above)
Comment (ikesau) [reply db/model/Gdoc/htmlToEnriched.ts:641]: Oof! Sorry about that! You're right, I removed it for the callout refactor and then forgot to change it back.

File: db/model/Gdoc/htmlToEnriched.ts (hunk 2 above)
Comment (ikesau) [reply db/model/Gdoc/htmlToEnriched.ts:620]: Yeah, we don't really have a way to render a similar component in Gdocs. Instead we'll track it and deal with it manually.

File: site/gdocs/ArticleBlock.tsx [hunk 3]
```diff
@@ -495,9 +495,23 @@ export default function ArticleBlock({
...
                     toc={toc}
-                    className={getLayout("sdg-toc", containerType)}
+                    title="List of targets and indicators"
+                    className={getLayout("toc", containerType)}
+                />
+            ) : null
+        })
+        .with({ type: "entry-summary" }, (block) => {
+            return toc ? (
+                <TableOfContents
```
Comment (mlbrgl) [inline site/gdocs/ArticleBlock.tsx:507]: I don't think this check makes sense here

**Review by mlbrgl — APPROVED** [review]
Looks good!

I'm not aware of a particular process to go through WP to Gdoc migration errors. Is there anything in place yet, that uses the info in the archieml_update_statistics column (other than the per doc "migration error" button)?

**Pull request merged**